[TensorExpr] Add a envvar to disable LLVM backend and use IR Eval instead.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48355 [TensorExpr] Add a envvar to disable LLVM backend and use IR Eval instead.**
* #48354 [TensorExpr] IREval: fix div for Half dtype.

Differential Revision: [D25139668](https://our.internmc.facebook.com/intern/diff/D25139668)

Commit:
[TensorExpr] Add a envvar to disable LLVM backend and use IR Eval instead.

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Add a envvar to disable LLVM backend and use IR Eval instead."


Differential Revision: [D25139668](https://our.internmc.facebook.com/intern/diff/D25139668)

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Add a envvar to disable LLVM backend and use IR Eval instead."


Differential Revision: [D25139668](https://our.internmc.facebook.com/intern/diff/D25139668)

[ghstack-poisoned]

nickgg reviewed (approved):
LGTM

eellison commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -50,6 +50,18 @@ bool fallbackEnforced() {
+  if (!enable_c_str) {
+    return false;
+  }
+  if (std::string(enable_c_str) == "1") {
```
nit: `return std::string(enable_c_str) == "1"`

eellison reviewed (approved):
nice

Commit:
Update on "[TensorExpr] Add a envvar to disable LLVM backend and use IR Eval instead."


Differential Revision: [D25139668](https://our.internmc.facebook.com/intern/diff/D25139668)

[ghstack-poisoned]
